packages/native/src/createStaticNavigation.tsx

```diff
@@ -63,6 +63,19 @@ export function createStaticNavigation(tree: StaticNavigation<any, any, any>) {
       return undefined;
     }, [linking?.config, linking?.enabled]);
 
+    const transformedLinking = React.useMemo(() => {
+      if (!linking) return undefined;
+
+      return {
+        ...linking,
+        enabled:
+          typeof linking.enabled === 'boolean'
+            ? linking.enabled
+            : screens != null,
+        config: screens ? { ...linking.config, screens } : undefined,
+      };
+    }, [linking, screens]);
```

Comment:
Looking at a bit more, we should be able to always keep same reference for `linking.config` with static config as it only uses 2 string properties from `linking.config`:

```js
const linkingConfig = React.useMemo(() => {
  const screens = tree.config.screens
    ? createPathConfigForStaticNavigation(
        tree,
        { initialRouteName: linking?.config?.initialRouteName },
        linking?.enabled === 'auto'
      )
    : undefined;

  const config = screens
    ? {
        path: linking?.config?.path,
        initialRouteName: linking?.config?.initialRouteName,
        screens,
      }
    : undefined;

  return config;
}, [
  linking?.enabled,
  linking?.config?.path,
  linking?.config?.initialRouteName,
]);

const memoizedLinking = React.useMemo(() => {
  if (!linking) {
    return undefined;
  }

  return {
    ...linking,
    enabled:
      typeof linking.enabled === 'boolean'
        ? linking.enabled
        : linkingConfig.screens != null,
    config: linkingConfig,
  };
}, [linking, screens]);
```

Reply:
@satya164 Thanks, suggestion applied [here](https://github.com/react-navigation/react-navigation/pull/12120/commits/74a71ad7d49ea3b9c83012cc13273d584086335b) ✅ 

I've refactored code a bit to take advantage of early returns and boost up readability. Let me know what you think!